extensions/amp-sidebar/1.0/amp-sidebar.js

```diff
@@ -103,13 +104,14 @@ export class AmpSidebar extends AMP.BaseElement {
       this.element.setAttribute('side', this.side_);
     }
 
+    const ampdoc = this.getAmpDoc();
     // Get the toolbar attribute from the child navs
     const toolbarElements =
-    Array.prototype.slice
-      .call(this.element.querySelectorAll('nav[toolbar]'), 0);
+      toArray(ampdoc.getRootNode().querySelectorAll('nav[toolbar]'));
```

Comment:
Sorry, I think I confused you. Your existing code with scoped `this.element.querySelectorAll` is good - because it's scoped to the element itself. The global selectors are the problem, e.g. `document.querySelector` or `document.getElementById` - those are the only once we need to be careful about (we are working on banning them altogether from code).